community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/Operations.java

```diff
@@ -514,16 +513,151 @@ public Value graphRemoveProperty( int propertyKey )
     }
 
+    @Override
+    public void nodeRemoveFromExplicitIndex( String indexName, long node, String key, Object value )
+            throws ExplicitIndexNotFoundKernelException
+    {
+        assertOpen();
+        ktx.explicitIndexTxState().nodeChanges( indexName ).remove( node, key, value );
+    }
+
+    @Override
+    public void nodeRemoveFromExplicitIndex( String indexName, long node, String key )
+            throws ExplicitIndexNotFoundKernelException
+    {
+        assertOpen();
+        ktx.explicitIndexTxState().nodeChanges( indexName ).remove( node, key );
+    }
+
+    @Override
+    public void nodeExplicitIndexCreate( String indexName, Map<String,String> customConfig )
+    {
+        assertOpen();
+        ktx.explicitIndexTxState().createIndex( IndexEntityType.Node, indexName, customConfig );
+    }
+
     @Override
     public void nodeExplicitIndexCreateLazily( String indexName, Map<String,String> customConfig )
     {
         assertOpen();
         allStoreHolder.getOrCreateNodeIndexConfig( indexName, customConfig );
     }
+
+    @Override
+    public void relationshipAddToExplicitIndex( String indexName, long relationship, String key, Object value )
+            throws EntityNotFoundException, ExplicitIndexNotFoundKernelException
+    {
+        throw new UnsupportedOperationException(  );
+    }
+
+    @Override
+    public void relationshipRemoveFromExplicitIndex( String indexName, long relationship, String key, Object value )
+            throws ExplicitIndexNotFoundKernelException, EntityNotFoundException
+    {
+        assertOpen();
+        ktx.explicitIndexTxState().relationshipChanges( indexName ).remove( relationship, key, value );
+    }
+
+    @Override
+    public void relationshipRemoveFromExplicitIndex( String indexName, long relationship, String key )
+            throws ExplicitIndexNotFoundKernelException, EntityNotFoundException
+    {
+        ktx.explicitIndexTxState().relationshipChanges( indexName ).remove( relationship, key );
+
+    }
+
+    @Override
+    public void relationshipRemoveFromExplicitIndex( String indexName, long relationship )
+            throws ExplicitIndexNotFoundKernelException, EntityNotFoundException
+    {
+        ktx.explicitIndexTxState().relationshipChanges( indexName ).remove( relationship );
+    }
+
+    @Override
+    public void relationshipExplicitIndexCreate( String indexName, Map<String,String> customConfig )
+    {
+        assertOpen();
+        ktx.explicitIndexTxState().createIndex( IndexEntityType.Relationship, indexName, customConfig );
+    }
+
+    @Override
+    public void relationshipExplicitIndexCreateLazily( String indexName, Map<String,String> customConfig )
+    {
+        assertOpen();
+        allStoreHolder.getOrCreateRelationshipIndexConfig( indexName, customConfig );
+    }
+
+    private Value readProperty( long node, int propertyKey ) throws EntityNotFoundException
```

Comment:
I think this should be called `readNodeProperty`, because I have a hunch that it'll get siblings in the future.

Reply:
nah, who needs relationships :)